### PR TITLE
Add new method `urlParameters` that takes a `Map`

### PR DESCRIPTION
**Pull request by brettwp**
Adding a new method `urlParameter(Map<String, Object> parameters)`.  I'm opening this as a draft PR to get some feedback.

### CODE REVIEW

File: src/main/java/com/inversoft/rest/RESTClient.java [hunk 1]
```diff
@@ -351,6 +351,20 @@ public URI getURI() {
...
   }
 
+  /**
+   * Add URL parameters from a {@code Map<String, Object>}.
+   *
+   * @param parameters The url parameters <code>Map</code> to add.  For each item in the <code>Map</code>
+   *                   this will call <code>urlParameter(String, Object)</code>
+   * @return This.
+   */
+  public RESTClient<RS, ERS> urlParameter(Map<String, Object> parameters) {
```
Comment (robotdan) [inline src/main/java/com/inversoft/rest/RESTClient.java]: I would vote to name this `urlParameters` (plural)

**Review by robotdan — COMMENTED** [review]
Looks good!

File: src/main/java/com/inversoft/rest/RESTClient.java (hunk 1 above)
Comment (voidmain) [reply src/main/java/com/inversoft/rest/RESTClient.java]: Yeah. That's the Java convention, which is different than other languages. Multiples using `s` to be grammatically aligned.

Comment (brettwp) [reply src/main/java/com/inversoft/rest/RESTClient.java]: Done.

**Comment by brettwp** [comment]
Last commit I bumped the version to `3.5.0`, changed `parameters` to `urlParameters`, and renamed the test method to better match the pattern `get_*`.  Also, changed draft PR to ready for review.

File: src/main/java/com/inversoft/rest/RESTClient.java [hunk 2]
```diff
@@ -351,6 +351,20 @@ public URI getURI() {
...
   }
 
+  /**
+   * Add URL parameters from a {@code Map<String, Object>}.
+   *
+   * @param urlParameters The url parameters <code>Map</code> to add.  For each item in the <code>Map</code>
+   *                      this will call <code>urlParameter(String, Object)</code>
+   * @return This.
+   */
+  public RESTClient<RS, ERS> urlParameter(Map<String, Object> urlParameters) {
```
Comment (robotdan) [inline src/main/java/com/inversoft/rest/RESTClient.java]: My comment was actually to rename the method itself, but the parameter is also a good change. 

Should be :

```
public RESTClient<RS, ERS> urlParameters(Map<String, Object> urlParameters) {
   // Code
}
```

**Review by robotdan — CHANGES_REQUESTED** [review]

File: src/main/java/com/inversoft/rest/RESTClient.java (hunk 2 above)
Comment (brettwp) [reply src/main/java/com/inversoft/rest/RESTClient.java]: Oops.  I misunderstood what var needed changing.  I just pushed an update for this.

**Review by robotdan — APPROVED** [review]

**Comment by robotdan** [comment]
We can test with this PR - I think now that I approved it, you have the option to merge it to master.

**Pull request merged**